gateway/sensor.go

```diff
@@ -101,13 +100,23 @@ func newGateway(
 	conf resource.Config,
 	logger logging.Logger,
 ) (sensor.Sensor, error) {
+	cfg, err := resource.NativeConfig[*Config](conf)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewGateway(ctx, conf.ResourceName().AsNamed(), cfg, logger)
+}
+
```

Comment:
```suggestion
```

Delete all the lines and export the constructor function. We shouldn't case the config from app in two places, particularly since we have a reconfigure flow and a build flow.